roachpb/data_test.go

```diff
@@ -695,3 +697,48 @@ func TestTransactionIDLen(t *testing.T) {
 		t.Fatalf("expected %d, got %d", TransactionIDLen, l)
 	}
 }
+
+func findPtrFields(t reflect.Type, prefix string) []string {
+	if t.Kind() != reflect.Struct {
+		return nil
+	}
+	var res []string
+	for i := 0; i < t.NumField(); i++ {
+		f := t.Field(i)
+		switch f.Type.Kind() {
+		case reflect.Ptr:
+			res = append(res, prefix+f.Name)
+			if f.Type.Elem().Kind() == reflect.Struct {
+				res = append(res, findPtrFields(f.Type.Elem(), f.Name+".")...)
+			}
+		case reflect.Slice:
+			res = append(res, prefix+f.Name)
+			res = append(res, findPtrFields(f.Type.Elem(), f.Name+".")...)
+		case reflect.Struct:
+			res = append(res, findPtrFields(f.Type, f.Name+".")...)
+		}
+	}
+	return res
+}
+
+func TestTransactionClone(t *testing.T) {
```

Comment:
We must be missing each others points somehow.
The problem with a botched clone is that you forget to make a copy (so you share memory with the original). Assuming a NoZeroField that indeed does recurse, what you're testing is making a copy in which nothing is zero, then making all the original memory that you copied from zero. If you are sharing memory with the original, you'll find something that's zero.
Maybe you need an exception or two (I think you're not cloning the key since you don't think it's going to change anyway), but those can be coded into the test.